src/main/java/core/driver/DriverFactory.java

```diff
@@ -0,0 +1,26 @@
+package main.java.core.driver;
+
+import main.java.core.listener.LoggingListener;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.support.events.EventFiringDecorator;
+
+public class DriverFactory {
+
+    public static WebDriver createBrowser(BrowserType browser) {
+        WebDriver driver;
+        switch (browser) {
```

Comment:
fyi i suggest implementing bonigarcia WebdriverManager lib to evade the need of manual drivers installation

`WebDriverManager.firefoxdriver().setup();
                driver = new FirefoxDriver();`